js/plates-view/plate-mesh-vertex.glsl

```diff
@@ -28,10 +34,14 @@ varying vec3 vViewPosition;
 #include <logdepthbuf_pars_vertex>
 #include <clipping_planes_pars_vertex>
 void main() {
-    // --- CUSTOM:
-    vColor = color;
-    vBumpScale = vertexBumpScale;
-    // ---
+
+  // --- CUSTOM:
+  vColor = color;
+  vBumpScale = vertexBumpScale;
+  vHidden = hidden;
+  vColormapValue = colormapValue;
+  // ---
+
```

Comment:
Again, I am not 100% sure how glsl vertex shaders work, but it seems like you are assigning values to the `varying` fields from the attributes(input?), which I guess get passed into the next step in the shader pipeline.  

I guess I am most puzzled by `transformed` assignment on 62, I don't see where that variable is being defined, nor `objectNormal`, those must defined in one of the `#include`s?  

The idea I think is to display the vertex by elevation towards its normal (away from the center of the sphere).

Reply:
> Again, I am not 100% sure how glsl vertex shaders work, but it seems like you are assigning values to the varying fields from the attributes(input?), which I guess get passed into the next step in the shader pipeline.

Yes. I mentioned that earlier, but this shader is run per each mesh vertex. The next fragment shader is run per each pixel. There much more pixels than vertices. So, when we assign an attribute to `varying`, it gets linearly interpolated. OpenGL will interpolate nicely these vertex attributes to fill all the space of the final, rendered triangle.

> I guess I am most puzzled by transformed assignment on 62, I don't see where that variable is being defined, nor objectNormal, those must defined in one of the #includes?

Yes, exactly, it's somewhere in the included pieces. This is a pretty hacky way to write shaders. I'm reusing THREE.JS PhongMaterial shader, and making just minimal custom changes. That way, I can reuse all the features of this material without rewriting the whole shader. Also, when THREE.JS and PhongMaterial get updated, it should be relatively easy to update this shader too.

> The idea I think is to display the vertex by elevation towards its normal (away from the center of the sphere).

Yes. That's a fast way to apply elevation to each vertex. Another approach is to simply modify position attributes.

Actually, I use both ways, depending on `config.flatShading`. 
The fast way done in a shader doesn't let us fix normals (as normals need to be calculated using a few vertices around and there's no access to that in shader code). So, geometry will be modified, but normals would be outdated, and lighting/shading will be incorrect (it'd actually look like there's no elevation). But if we use `flatShading` mode, normals are not necessary, and then this method is good enough.

If we don't use `flatShading`, I use the slow way - I update position attributes in JS code, and then recalculate normals all the time. It ensures that lighting is also correct, but is a bit slower.